library/std/src/sys/wasi/mod.rs

```diff
@@ -83,24 +83,82 @@ pub fn decode_error_kind(errno: i32) -> std_io::ErrorKind {
+        e if e == wasi::ERRNO_ISDIR.raw().into() => IsADirectory,
+        e if e == wasi::ERRNO_LOOP.raw().into() => FilesystemLoop,
+        //        wasi::ERRNO_MFILE.raw().into() => "file descriptor value too large",
+        e if e == wasi::ERRNO_MLINK.raw().into() => TooManyLinks,
+        //        wasi::ERRNO_MSGSIZE.raw().into() => "message too large",
+        //        wasi::ERRNO_MULTIHOP.raw().into() => "reserved",
+        e if e == wasi::ERRNO_NAMETOOLONG.raw().into() => InvalidFilename,
+        e if e == wasi::ERRNO_NETDOWN.raw().into() => NetworkDown,
+        //        wasi::ERRNO_NETRESET.raw().into() => "connection aborted by network",
+        e if e == wasi::ERRNO_NETUNREACH.raw().into() => NetworkUnreachable,
+        //        wasi::ERRNO_NFILE.raw().into() => "too many files open in system",
+        //        wasi::ERRNO_NOBUFS.raw().into() => "no buffer space available",
+        e if e == wasi::ERRNO_NODEV.raw().into() => NotFound,
+        e if e == wasi::ERRNO_NOENT.raw().into() => NotFound,
+        e if e == wasi::ERRNO_NOEXEC.raw().into() => InvalidData,
+        //        wasi::ERRNO_NOLCK.raw().into() => "no locks available",
+        //        wasi::ERRNO_NOLINK.raw().into() => "reserved",
         e if e == wasi::ERRNO_NOMEM.raw().into() => OutOfMemory,
+        //        wasi::ERRNO_NOMSG.raw().into() => "no message of the desired type",
+        //        wasi::ERRNO_NOPROTOOPT.raw().into() => "protocol not available",
+        e if e == wasi::ERRNO_NOSPC.raw().into() => StorageFull,
+        e if e == wasi::ERRNO_NOSYS.raw().into() => Unsupported,
+        e if e == wasi::ERRNO_NOTCONN.raw().into() => NotConnected,
+        e if e == wasi::ERRNO_NOTDIR.raw().into() => NotADirectory,
+        e if e == wasi::ERRNO_NOTEMPTY.raw().into() => DirectoryNotEmpty,
+        //        wasi::ERRNO_NOTRECOVERABLE.raw().into() => "state not recoverable",
+        e if e == wasi::ERRNO_NOTSOCK.raw().into() => InvalidInput,
+        e if e == wasi::ERRNO_NOTSUP.raw().into() => Unsupported,
+        //        wasi::ERRNO_NOTTY.raw().into() => "inappropriate I/O control operation",
+        e if e == wasi::ERRNO_NXIO.raw().into() => NotFound,
+        //        wasi::ERRNO_OVERFLOW.raw().into() => "value too large to be stored in data type",
+        //        wasi::ERRNO_OWNERDEAD.raw().into() => "previous owner died",
+        e if e == wasi::ERRNO_PERM.raw().into() => PermissionDenied,
+        e if e == wasi::ERRNO_PIPE.raw().into() => BrokenPipe,
+        //        wasi::ERRNO_PROTO.raw().into() => "protocol error",
+        e if e == wasi::ERRNO_PROTONOSUPPORT.raw().into() => Unsupported,
+        e if e == wasi::ERRNO_PROTOTYPE.raw().into() => InvalidInput,
+        //        wasi::ERRNO_RANGE.raw().into() => "result too large",
+        e if e == wasi::ERRNO_ROFS.raw().into() => ReadOnlyFilesystem,
+        //        wasi::ERRNO_SPIPE.raw().into() => "invalid seek",
```

Comment:
Doublecheck the commented-out errors for `io_error_more` variants, please?
```suggestion
        e if e == wasi::ERRNO_SPIPE.raw().into() => NotSeekable,
```

Reply:
I read "invalid seek" as "out of range", but seems like `spipe` stands for `SeekPipe`, hence the invalid seek.

I checked all the variants, should be ok now.